### ISSUE TITLE
Bytecode changes for dict in python 3.6 causes batavia.core.Dict to fail

### ISSUE THREAD

**Issue by slick666** [pybee/batavia#181]
This is a continuation of #26 that was fixed in 3.4 and 3.5.. Python 3.6 fails for this same input but for different reasons. 

With python 3.5 the bytecode for storing data to the dict has changed to:

  2           0 LOAD_CONST               1 ('a')
              3 LOAD_CONST               2 (1)
              6 LOAD_CONST               3 ('b')
              9 LOAD_CONST               4 (2)
             12 BUILD_MAP                2
             15 STORE_FAST               0 (lonk)
causing dicts to be empty.

To reproduce use (in the testbed):

foo = {'a': 1, 'b': 2}
print('bar', foo)
Expected result: bar 1
Actual result: bar

**Comment by cjrh** [comment]
Some exploration for 3.6:

```python
>>> code = '''\
... foo = {'a': 1, 'b': 2}
... print('bar', foo)'''
>>> code
"foo = {'a': 1, 'b': 2}\nprint('bar', foo)"
>>> cobj = compile(code, 'blah', 'exec')
>>> cobj
<code object <module> at 0x10889d420, file "blah", line 1>
>>> import dis
>>> dis.dis(cobj)
  1           0 LOAD_CONST               0 (1)
              2 LOAD_CONST               1 (2)
              4 LOAD_CONST               2 (('a', 'b'))
              6 BUILD_CONST_KEY_MAP      2
              8 STORE_NAME               0 (foo)

  2          10 LOAD_NAME                1 (print)
             12 LOAD_CONST               3 ('bar')
             14 LOAD_NAME                0 (foo)
             16 CALL_FUNCTION            2
             18 POP_TOP
             20 LOAD_CONST               4 (None)
             22 RETURN_VALUE
```

**Comment by cjrh** [comment]
There are a bunch of new opcodes, so I'm adding these to `batavia/batavia/modules/dis.js`.  I'll also add (empty to start) implementations in `batavia/batavia/VirtualMachine.js`.

**Comment by cjrh** [comment]
I've implemented the `BUILD_CONST_KEY_MAP` javasscript code (in the virtual machine) for Python 3.6 bytecode, in PR #536.  However, due to the wordcode change, there is still a lot of other things that are not working.

**Comment by cjrh** [comment]
Note that in PR #536, this snippet:

```
foo = {'a': 1, 'b': 2}
print('bar', foo)
```

Now does actually work in the Batavia testbed "run python in the browser".  However, more work needs to be done in #536, particularly making the tests pass.

**Comment by Zac-HD** [comment]
Also closed by #621 I think 😉

**Comment by freakboy3742** [comment]
Indeed!
